src/classes/Pricer.ts

```diff
@@ -0,0 +1,138 @@
+export interface PricerOptions {
+    pricerUrl?: string;
+    pricerApiToken?: string;
+}
+
+export type GetPricerFn = (options: PricerOptions) => Pricer;
+
+/**
+ * Basic pricer interface
+ *
+ * Those wishing to plug in something other than PricesTf should provide at least
+ * the following interface plus a static getPricer(options: PricerOptions): Pricer
+ * to get the constructor
+ */
+export default interface Pricer {
+    getOptions(): PricerOptions;
+
+    requestCheck(sku: string, source: string): Promise<RequestCheckResponse>;
+
+    getPrice(sku: string, source: string): Promise<GetItemPriceResponse>;
+
+    getSales(sku: string, source: string): Promise<GetItemSalesResponse>;
+
+    getLinks(sku: string): Promise<GetItemLinks>;
+
+    getPriceHistory(sku: string, source: string): Promise<GetItemHistoryResponse>;
+
+    getPricelist(source: string): Promise<GetPricelistResponse>;
+
+    getSchema(): Promise<GetSchemaResponse>;
+}
+
+export type RequestCheckFn = (sku: string, source: string) => Promise<RequestCheckResponse>;
+export type GetPriceFn = (sku: string, source: string) => Promise<GetItemPriceResponse>;
+export type GetSalesFn = (sku: string, source: string) => Promise<GetItemSalesResponse>;
+export type GetLinksFn = (sku: string) => Promise<GetItemLinks>;
+export type GetPriceHistoryFn = (sku: string, source: string) => Promise<GetItemHistoryResponse>;
+export type GetPrice = (sku: string, source: string) => Promise<GetItemPriceResponse>;
+export type GetPricelist = (source: string) => Promise<GetPricelistResponse>;
+export type GetSchema = () => Promise<GetSchemaResponse>;
+
+export interface PricesResponse {
+    success: boolean;
+    message?: string;
+}
+
+export interface GetSchemaResponse extends PricesResponse {
+    version: string;
+    time: number;
+    raw: any;
+}
+
+export interface GetOverviewResponse extends PricesResponse {
+    items: ItemOverview[];
+}
+
+export interface ItemOverview {
+    name: string;
+    sku: string;
+}
+
+export interface GetPricelistResponse extends PricesResponse {
+    currency?: any;
+    items?: Item[];
+}
+
+export interface Item {
+    sku: string;
+    name: string;
+    source: string;
+    time: number;
+    buy: KeysMetal | null;
+    sell: KeysMetal | null;
+}
+
+export interface KeysMetal {
+    keys: number;
+    metal: number;
+}
```

Comment:
This can be replaced by `Currencies` by importing it from `tf2-currencies-2` library.